contracts/mainnet/connectors/morpho-blue/helpers.sol

```diff
@@ -0,0 +1,193 @@
+			? _marketParams.collateralToken == ethAddr
+			: _marketParams.loanToken == ethAddr;
+
+		_marketParams = updateTokenAddresses(_marketParams);
+
+		// Check for max value
+		if (_assets == type(uint256).max) {
+			uint256 _maxAvailable = _isEth
+				? address(this).balance
+				: TokenInterface(_marketParams.loanToken).balanceOf(
+					address(this)
+				);
+			if (_mode == Mode.Repay) {
+				uint256 _amtDebt = getPaybackBalance(_marketParams, _onBehalf);
+				_amt = min(_maxAvailable, _amtDebt);
+			} else {
+				_amt = _maxAvailable;
+			}
+		}
+
+		// Perform conversion if necessary
+		if (_isEth) {
+			convertEthToWeth(
+				true,
+				TokenInterface(_marketParams.loanToken),
+				_amt
+			);
+		}
+
+		return (_marketParams, _amt);
+	}
+
+	/// @notice Handles Eth to Weth conversion if shares are provided.
+	function _performEthToWethSharesConversion(
+		MarketParams memory _marketParams,
+		uint256 _shares,
+		address _onBehalf,
+		uint256 _getId,
+		bool _isRepay
+	) internal returns (MarketParams memory, uint256 _assets) {
+		uint256 _shareAmt = getUint(_getId, _shares);
+		bool _isEth = _marketParams.loanToken == ethAddr;
+
+		_marketParams = updateTokenAddresses(_marketParams);
+
+		// Handle the max share case
+		if (_shares == type(uint256).max) {
+			uint256 _maxAvailable = _isEth
+				? address(this).balance
+				: TokenInterface(_marketParams.loanToken).balanceOf(
+					address(this)
+				);
+
+			// If it's repay calculate the min of balance available and debt to repay
+			if (_isRepay) {
+				_assets = min(
+					_maxAvailable,
+					getPaybackBalance(_marketParams, _onBehalf)
+				);
+			} else {
+				_assets = _maxAvailable;
+			}
+		} else {
+			bytes32 _id = id(_marketParams);
+			_assets = _toAssetsUp(
+				_shareAmt,
+				MORPHO_BLUE.market(_id).totalSupplyAssets,
+				MORPHO_BLUE.market(_id).totalSupplyShares
+			);
+		}
+
+		// Perform ETH to WETH conversion if necessary
+		if (_isEth) {
+			convertEthToWeth(
+				true,
+				TokenInterface(_marketParams.loanToken),
+				_assets
+			);
+		}
+
+		return (_marketParams, _assets);
+	}
+
+	/// @notice Helper function to find the minimum of two values
+	function min(uint256 a, uint256 b) internal pure returns (uint256) {
+		return a < b ? a : b;
+	}
+
+	/// @notice Returns the payback balance in assets.
+	function getPaybackBalance(
+		MarketParams memory _marketParams,
+		address _onBehalf
+	) internal view returns (uint256 _assets) {
+		bytes32 _id = id(_marketParams);
+
+		uint256 _shareAmt = MORPHO_BLUE.position(_id, _onBehalf).supplyShares;
+
+		_assets = _toAssetsUp(
+			_shareAmt,
+			MORPHO_BLUE.market(_id).totalSupplyAssets,
+			MORPHO_BLUE.market(_id).totalSupplyShares
+		);
+	}
+
+	/// @notice Returns the id of the market `marketParams`.
+	function id(
+		MarketParams memory marketParams
+	) internal pure returns (bytes32 marketParamsId) {
+		assembly {
+			marketParamsId := keccak256(
+				marketParams,
+				MARKET_PARAMS_BYTES_LENGTH
+			)
+		}
+	}
```

Comment:
This could use our audited `MarketParamsLib`: https://github.com/morpho-org/morpho-blue/blob/414b67a523dbc72144a5dfed62ac60771f15d98e/src/libraries/MarketParamsLib.sol#L15-L20

Reply:
Updated! Same reason as [this](https://github.com/Instadapp/dsa-connectors/pull/321/files#r1428850264)